Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -43,7 +43,10 @@ void GridMasonryLayout::performMasonryPlacement(unsigned gridAxisTracks, GridTra
     m_renderGrid.currentGrid().setupGridForMasonryLayout();
     m_renderGrid.populateExplicitGridAndOrderIterator();
 
-    resizeAndResetRunningPositions(); 
+    resizeAndResetRunningPositions();
+
+    m_renderGrid.populateGridPositionsForDirection(ForColumns);
+    m_renderGrid.populateGridPositionsForDirection(ForRows);
```

Comment:
I think we only need to do this for the grid-axis direction, since that's the one we did sizing for. The masonry axis doesn't really have tracks, and we haven't done any sort of 'sizing', so copying across positions for it is a bit meaningless.

Reply:
I probably want some code here to track which direction is the masonry axis.